### PR TITLE
`list_lsps`: Skip LSPs that fail validation

### PR DESCRIPTION
**Pull request by ok300**
- Validate `LspInformation` when initializing
  - Validation consists of checking if the `opening_fee_params_menu` is valid
- `BreezServices::list_lsps()` only returns entries that pass validation

### CODE REVIEW

**Review by erdemyerebasmaz — APPROVED** [review]
LGTM

File: libs/sdk-core/src/lsp.rs [hunk 1]
```diff
@@ -56,7 +57,10 @@ impl LspInformation {
                 .into_iter()
                 .map(|ofp| ofp.into())
                 .collect::<Vec<OpeningFeeParams>>(),
-        }
+        };
+
+        OpeningFeeParamsMenu::try_from(info.opening_fee_params_menu.clone())?;
```
Comment (roeierez) [inline libs/sdk-core/src/lsp.rs:62]: It looks a bit weird that we first convert the opening_fee_params to our own structure and then validate. I is more natural flow to validate the raw input and then convert it.
If this is the only point we need to validate then maybe move the validate to this struct implementation?

**Review by JssDWt — APPROVED** [review]

File: libs/sdk-core/src/lsp.rs (hunk 1 above)
Comment (ok300) [reply libs/sdk-core/src/lsp.rs]: Addressed in the latest commit:

https://github.com/breez/breez-sdk/pull/292/commits/8e3d706a1eea7a6a728d4514cea8620ca23086ce#diff-209db7ea9780093f9c880649e7f40a90a4615da7cee3111a299bbfa430b3aec2L55-L62

Part of it was also:
- changing `LspInformation.opening_fee_params_menu` type to `OpeningFeeParamsMenu`, so the menu is validated on instantiation
- moved `OpeningFeeParamsMenu` to `models.rs`
- updated uniffi bindings

File: libs/sdk-core/src/lsp.rs [hunk 2]
```diff
@@ -51,33 +52,42 @@ impl LspInformation {
...
             max_inactive_duration: lsp_info.max_inactive_duration,
             channel_minimum_fee_msat: lsp_info.channel_minimum_fee_msat,
-            opening_fee_params_menu: lsp_info
-                .opening_fee_params_menu
-                .into_iter()
-                .map(|ofp| ofp.into())
-                .collect::<Vec<OpeningFeeParams>>(),
-        }
+            opening_fee_params_menu: OpeningFeeParamsMenu::try_from(
+                lsp_info
```
Comment (roeierez) [inline libs/sdk-core/src/lsp.rs:56]: This looks a lot better!
Perhaps also it is better to change the try_from to get the grpc values and move the mapping into our model inside that function?

Comment (ok300) [reply libs/sdk-core/src/lsp.rs]: Done

**Review by roeierez — APPROVED** [review]
LGTM

**Comment by roeierez** [comment]
@ok300 I am guessing the buymoonpay test fails because it uses the mock LSP which returns the old structure. Now that the code validates the LSP list it probably returned empty list.

**Comment by ok300** [comment]
@roeierez turns out it was a combination of
- using the default Staging LSP ID, but using an LSP mock with a test ID of `1`
- getting a `None` dyanmic fee rate in `receive_onchain` because the user fee was `None` and the mocked LSP fee was not at least 48h valid

Fixed in 073499a41303b34edbe8ecc83e4816a22b021a01

**Review by erdemyerebasmaz — COMMENTED** [review]
Enum changes look good 5c0bb3c90611cce104c1b4ce5d331ed724ae2bc3

File: libs/sdk-core/src/swap.rs [hunk 3]
```diff
@@ -408,7 +395,7 @@ impl BTCReceiveSwap {
                     amount_sats: swap_info.confirmed_sats,
                     description: String::from("Bitcoin Transfer"),
                     preimage: Some(swap_info.preimage),
-                    opening_fee_params: swap_info.channel_opening_fees,
+                    opening_fee_params: None,
```
Comment (JssDWt) [inline libs/sdk-core/src/swap.rs]: Shouldn't this be passed in here?

Comment (ok300) [reply libs/sdk-core/src/swap.rs]: I checked, this is based on a previous commit. The param is currently passed correctly.

**Comment by ok300** [comment]
As discussed, merging this into `dynamic_fees`, so we can use that as the basis for the full dynamic fees review.

**Pull request merged**